hearthstone/simulator/core/hero_pool.py

```diff
@@ -308,7 +308,7 @@ class ArchVillianRafaam(Hero):
     base_power_cost = 1
 
     def handle_event_powers(self, event: 'CardEvent', context: Union['BuyPhaseContext', 'CombatPhaseContext']):
-        if event.event is EVENTS.DIES and event.card in context.enemy_war_party.board and self.hero_power_used:
+        if event.event is EVENTS.DIES and event.card in context.enemy_war_party.board and self.power_uses_left_this_turn == 0:
```

Comment:
I think counting up is better than counting down, because conditions like this are counterintuitive.